.github/workflows/build_test_linux_fedora.yaml

```diff
@@ -27,7 +27,7 @@ jobs:
     - name: Build dpsimpy and test examples
       shell: bash
       working-directory: ${{ github.workspace }}/build
-      run: cmake --build . --target dpsimpy --target tests --target dpsimpyvillas -j`nproc`
+      run: cmake --build . --target dpsimpy --target tests --target dpsimpyvillas -j$(cat /proc/cpuinfo|grep -w processor|wc -l)
```

Comment:
Hey @leonardocarreras,

could you briefly describe the motivation behind using `/proc/cpuinfo` vs `nproc` here?

I believe, if we simply pass -j, we should already use maximum parallelism. Because of:

CMake ([docs(https://cmake.org/cmake/help/latest/manual/cmake.1.html#cmdoption-cmake-build-j)):
> If <jobs> is omitted the native build tool's default number is used.

GNU Make:
> If the -j option is given without an argument, make will not limit the number of jobs that can run simultaneously.

Reference: https://www.flamingspork.com/blog/2020/11/25/why-you-should-use-nproc-and-not-grep-proc-cpuinfo/

A little side note: I would prefer it if we always use long options (`--parallel` vs `-j` in scripts as its more expressive. Short options are great if you use them on the CLI. In scripts we want the code to be self-documenting as much as possible. Hence using a long name is easier to understand and review..

Reply:
Thanks @stv0g for having a look!

This is WIP, I just had some errors in some runs and I was trying options. The errors were due to other factors.

I will take then the `--parallel` option to make it easier to read.